### ISSUE TITLE
[Backport][2.7] Support k8s 1.27

### ISSUE THREAD

**Issue by manno** [rancher/fleet#2084]
Refers to rancher/rancher#44098, rancher/rancher#44127

**Comment by sbulage** [comment]
### System information:
- Rancher version: 2.7.11-head (fc2c10a)
- Fleet version: v0.8.2-rc.3

###  Rancher Upgrade Scenario

- [ ]  Upgrade from Rancher `2.7.10` to `2.7.11` works fine.

### Fresh Rancher installation

- [ ]  Fresh Rancher installation with the `v1.27.1+k3s1` or beyond.

**Comment by mmartin24** [comment]
> ### System information:
> * Rancher version: 2.7.11-head (fc2c10a)
> * Fleet version: v0.8.2-rc.3
> 
> ### Rancher Upgrade Scenario
> * [ ]   Upgrade from Rancher `2.7.10` to `2.7.11` works fine.
> 
> ### Fresh Rancher installation
> * [ ]   Fresh Rancher installation with the `v1.27.1+k3s1` or beyond.

Checked that fresh install was not possible with Kubernetes `v1.27.6+k3s1`

Error:
```
Error: INSTALLATION FAILED: chart requires kubeVersion: < 1.27.0-0 which is incompatible with Kubernetes v1.27.6+k3s1
```
@gunamata would you pls be so kind to check it?

**Comment by sbulage** [comment]
The installation of Rancher is blocked on K8S 1.27.x.
Another issue: https://github.com/rancher/rancher/issues/44141#issuecomment-1939227197

**Comment by sbulage** [comment]
## System Information.
Rancher Version: 2.7.11-head (fc2c10a)
Fleet Version: 0.8.2-rc.3

## Below tests were performed:
- [ ] Update K8s version from v1.26.10  to 1.27.8 on local cluster
- [ ] Update K8s version from v1.26.10  to 1.27.8 on Imported cluster
- [ ] Update K8s version from v1.26.10  to 1.27.8 on both clusters

In both the cases, fleet repository was created before the update of K8S version.

**Comment by thehejik** [comment]
## Report for upgrade k8s and rancher
Testing upgrade of k8s 1.26 -> 1.27 and upgrade of `rancher:2.7.11-head`  (fc2c10a) to `rancher:2.8.2`

* started with rancher 2.7.11-head (fc2c10a), Fleet Version: 0.8.2-rc.3 on upstream RKE2 `v1.26.12+rke2r1` & with imported K3S `v1.26.10+k3s1`
* deployed testing fleet workload on `fleet-local` and `fleet-default` workspaces
* verified fleet images:
```
# upstream
$ for i in {cattle-fleet-system,cattle-fleet-local-system}; do k get pods -n $i -o jsonpath='{.items[*].spec.containers[*].image}'; echo; done
rancher/fleet:v0.8.2-rc.3 rancher/gitjob:v0.8.1
rancher/fleet-agent:v0.8.2-rc.3

# downstream
$ for i in {cattle-fleet-system,cattle-fleet-local-system}; do k get pods -n $i -o jsonpath='{.items[*].spec.containers[*].image}'; echo; done
rancher/fleet-agent:v0.8.2-rc.3
```
* upstream upgraded to `v1.27.10+rke2r1` over webui - fleet and its workload ok
* downstream upgraded to `v1.27.10+k3s2 ` over webui - fleet and its workload ok
* upgrade rancher to 2.8.2 - fleet has been upgraded and its workload ok
* verified fleet images:
```
# upstream 
$ for i in {cattle-fleet-system,cattle-fleet-local-system}; do k get pods -n $i -o jsonpath='{.items[*].spec.containers[*].image}'; echo; done
rancher/fleet:v0.9.0 rancher/gitjob:v0.1.96
rancher/fleet-agent:v0.9.0

# downstream
$ for i in {cattle-fleet-system,cattle-fleet-local-system}; do k get pods -n $i -o jsonpath='{.items[*].spec.containers[*].image}'; echo; done
rancher/fleet-agent:v0.9.0
```
* No traces of any issue in the rancher update jobs for fleet

**Comment by thehejik** [comment]
## System Info
Fleet App version on rancher:2.7.11-alpha3: `fleet[-crd]:102.2.2+up0.8.2-rc.4`:
(using staging registry and optimus repository for Rancher)
```bash
$ for i in {cattle-fleet-system,cattle-fleet-local-system,fleet-local}; do k get pods -n $i -o jsonpath='{.items[*].spec.containers[*].image}'; echo; done
rancher/fleet:v0.9.0
rancher/gitjob:v0.1.96
rancher/fleet-agent:v0.9.0
```

## Report for rancher upgrade on k3s v1.27
Testing upgrade of `rancher:2.7.11-alpha3` to `rancher:2.8.2` on k3s `v1.27.10+k3s2` (both local and downstream). 

- [x] Upgrade of `rancher:2.7.11-alpha3` to `rancher:2.8.2` ok with added fleet workload:
- private BitBucket HTTPS repo in fleet-default
- private Azure DevOps HTTPS in fleet-default (unable to "git fetch" on upgraded rancher:2.8.2 as expected)
- public HTTPS GitHub testing traefik chart functionality in fleet-local

Fleet App version on 2.8.2 after upgrade: `fleet[-crd]:103.1.0+up0.9.0`

All good, no traces of any issue in the rancher installation neither in the upgrade jobs for fleet, workload survived (apart Azure DevOps repo as expected).

Closing as completed.
